pkg/plugins/builtin/subtask/subtask.go

```diff
@@ -112,27 +115,39 @@ func exec(stepName string, config interface{}, ctx interface{}) (interface{}, in
 			return nil, nil, err
 		}
 
-		var resolverUsernames, watcherUsernames []string
+		var resolverUsernames, resolverGroups, watcherUsernames, watcherGroups []string
 		if cfg.ResolverUsernames != "" {
 			resolverUsernames, err = utils.ConvertJSONRowToSlice(cfg.ResolverUsernames)
 			if err != nil {
 				return nil, nil, fmt.Errorf("can't convert JSON to row slice: %s", err)
 			}
 		}
+		if cfg.ResolverGroups != "" {
+			resolverGroups, err = utils.ConvertJSONRowToSlice(cfg.ResolverGroups)
+			if err != nil {
+				return nil, nil, fmt.Errorf("can't convert JSON to row slice: %s", err)
+			}
+		}
 		if cfg.WatcherUsernames != "" {
 			watcherUsernames, err = utils.ConvertJSONRowToSlice(cfg.WatcherUsernames)
 			if err != nil {
 				return nil, nil, fmt.Errorf("can't convert JSON to row slice: %s", err)
 			}
 		}
+		if cfg.WatcherGroups != "" {
+			watcherGroups, err = utils.ConvertJSONRowToSlice(cfg.WatcherGroups)
+			if err != nil {
+				return nil, nil, fmt.Errorf("can't convert JSON to row slice: %s", err)
+			}
+		}
 
 		// TODO inherit watchers from parent task
 		ctx := auth.WithIdentity(context.Background(), stepContext.RequesterUsername)
```

Comment:
We have an issue here: in case the µTask instance is started with group support, this identity context will not inherit the groups of the user, because we are completely disconnected from the original HTTP request.
We need to find a way to propagate the groups here, otherwise, the call to `taskutils.CreateTask` will not behave correctly.

Reply:
Yes, that was exactly the solution I thought about.
Personally, I'm not a fan, but I can't think to any other solutions so let's do that 👍